src/Sentry/Protocol/Device.cs

```diff
@@ -456,7 +453,14 @@ public static Device FromJson(JsonElement json)
         var bootTime = json.GetPropertyOrNull("boot_time")?.GetDateTimeOffset();
         var processorCount = json.GetPropertyOrNull("processor_count")?.GetInt32();
         var cpuDescription = json.GetPropertyOrNull("cpu_description")?.GetString();
-        var processorFrequency = json.GetPropertyOrNull("processor_frequency")?.GetInt32();
+
+        // TODO: For next major: Remove this and change ProcessorFrequency from int to float
+        // The Java SDK reports the processorFrequency as `double`
+        // Java https://github.com/getsentry/sentry-java/blob/9762f09afa51944b40a9b77e116a55e54636e6c5/sentry/src/main/java/io/sentry/protocol/Device.java#L130
+        var processorFrequency = json.GetPropertyOrNull("processor_frequency")?.TryGetDouble(out var frequency) == true
```

Comment:
Should we use pattern matching instead of `==` operator?
```suggestion
        var processorFrequency = json.GetPropertyOrNull("processor_frequency")?.TryGetDouble(out var frequency) is true
```